RealmSwift/Tests/CombineTests.swift

```diff
@@ -190,6 +190,48 @@ class CombineRealmTests: CombinePublisherTestCase {
         }
         wait(for: [exp], timeout: 1)
     }
+
+    func testAsyncWrite() throws {
+        let exp = XCTestExpectation()
+
+        let intObject = SwiftIntObject()
+        let boolObject = SwiftBoolObject()
+        let stringObject = SwiftStringObject()
+
+        let listObject = SwiftListObject()
+
+        try! realm.write {
+            realm.add([intObject, boolObject, stringObject, listObject])
+        }
+
+        XCTAssertTrue(Thread.isMainThread)
+
+        cancellable = self.realm.asyncWrite(on: subscribeOnQueue, intObject, boolObject) { realm, intObject, boolObject in
+            XCTAssertFalse(Thread.isMainThread)
+            intObject.intCol = 42
+            boolObject.boolCol = true
+        }.combineWrite(on: receiveOnQueue, with: listObject.double, stringObject) { realm, doubleList, stringObject in
```

Comment:
Just looking at the API (without any idea about combine), wouldn't it be desirable to have a to pass a result from `asyncWrite` to `combineWrite`? E.g. something like:

```swift
self.realm.asyncWrite(on: queue1) { realm in
    let newObject = SwiftBoolObject()
    newObject.boolCol = true
    realm.add(newObject)
    return newObject
}.combineWrite(on: queue2) { realm, boolObject in
    // boolObject here is the value returned from asyncWrite

    if (boolObject.boolCol) {
        // blah blah
    }
}
```

Reply:
I thought about that– Thomas also mentioned to me that `combineWrite` doesn't seem to make sense to him, considering that write transactions cannot occur at the same time anyways.

It would be fairly trivial to pass objects down the chain with Combine, but I couldn't think of a good use case. Even in your use case above, it's technically pointless– whatever operation you are doing can just be done after the `realm.add`.

Reply:
Agree - combineWrite seems to be a bit niche. The only kind-of-sort-of valid use case I can think of is if you want to dispatch the second write to the UI thread because you need to access something UIKit-y, although would be hard pressed to find a realistic example.